src/components/wells/Sample.vue

```diff
@@ -96,6 +92,7 @@ export default {
           if (this.concentration < qcResults.warningThreshold.value) {
             this.warning = true
```

Comment:
If you have warning as an object you can just do `this.warning = { ...qcResults.warningThreshold }` and then your v-if could change to warning.message. I think that would mean you would not need the boolean.